gateway-service/src/main/java/org/zowe/apiml/gateway/ribbon/ApimlRibbonRetryFactory.java

```diff
@@ -10,57 +10,67 @@
 
 package org.zowe.apiml.gateway.ribbon;
 
-import lombok.RequiredArgsConstructor;
-import lombok.experimental.Delegate;
 import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryContext;
 import org.springframework.cloud.client.loadbalancer.LoadBalancedRetryPolicy;
 import org.springframework.cloud.client.loadbalancer.ServiceInstanceChooser;
 import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancedRetryFactory;
+import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancedRetryPolicy;
+import org.springframework.cloud.netflix.ribbon.RibbonLoadBalancerContext;
 import org.springframework.cloud.netflix.ribbon.SpringClientFactory;
+import org.springframework.http.HttpMethod;
 import org.springframework.retry.RetryListener;
 
+import java.lang.reflect.InvocationTargetException;
+import java.net.ConnectException;
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Allows adding RetryListeners to Ribbon Retry
  */
 public class ApimlRibbonRetryFactory extends RibbonLoadBalancedRetryFactory {
```

Comment:
Seems this class is missing unit tests, now that it has more logic it should have